__tests__/workers/cdc/primary.ts

```diff
@@ -3441,6 +3442,154 @@ describe('user streak change', () => {
       expect(alert.showRecoverStreak).toEqual(true);
     });
 
+    it('should set cache of previous streak if it has only been 2 valid days even when less than 48 hours', async () => {
+      jest
+        .useFakeTimers({ doNotFake })
+        .setSystemTime(new Date('2024-09-04T07:15:10')); // Wednesday
+      const lastViewAt = new Date('2024-09-02T10:25:27'); // Monday
+      const after: ChangeObject<ObjectType> = {
+        ...base,
+        lastViewAt: lastViewAt.toISOString() as never,
+        currentStreak: 0,
+      };
+      await con
+        .getRepository(UserStreak)
+        .update({ userId: '1' }, { lastViewAt });
+      await expectSuccessfulBackground(
+        worker,
+        mockChangeMessage<ObjectType>({
+          after,
+          before: {
+            ...base,
+            currentStreak: 3,
+            lastViewAt: lastViewAt.toISOString() as never,
+          },
+          op: 'u',
+          table: 'user_streak',
+        }),
+      );
+      const lastStreak = await getRestoreStreakCache({ userId: after.userId });
+      expect(triggerTypedEvent).toHaveBeenCalledTimes(1);
+      expect(jest.mocked(triggerTypedEvent).mock.calls[0].slice(1)).toEqual([
+        'api.v1.user-streak-updated',
+        { streak: after },
+      ]);
+      expect(lastStreak).toEqual(3);
+      const alert = await con.getRepository(Alerts).findOneBy({ userId: '1' });
+      expect(alert.showRecoverStreak).toEqual(true);
```

Comment:
I think for a complicated feature like this one it's better to use a describe block with tests for each of the expect test  (for each of the ones that make sense as a final result)- would make it easier for debugging in case one of these things only starts failing - like cache is set but alert is not set to true. 
This makes the title of the test a bit more clear, as for the moment it mentions only the cache but it's also doing a bunch of other things we need like setting the alert to true.

What I think would help is:

```
describe('Reset streak if it has only been 2 valid days even when less than 48 hours'){
beforeEach(... if we can do all the computations we need before here..)

it('should set cache of previous streak to ...')
it('should set alert to showRecoverStreak to true')
...
}
```

Reply:
It can set one or the other only. If the cache is not set, then the alert was not set. Vice versa, so it wouldn't be too beneficial to have them separated. This would just mean we will multiply our tests by two.